### DISCUSSION TITLE
🔗 Enable item description in Nav Menu widget

### DISCUSSION THREAD

**Discussion by enumans** [elementor/elementor#18859] [Feature Request]
Wordpress allows you to add a description to a menu item. This is standard Wordpress functionality. The Nav Menu Widget (Pro) does not support showing this description however. 

Can the Nav Menu Widget be extended so that the menu item description is also displayed? (It would be sufficient if it's simply rendered in HTML, styling can be done with CSS.)

**Comment by jp-infinite-digital-solutions** [comment]
Until this is added as a feature on elementor, you can use this code snippet on your child theme functions.php file to enable it:
`add_filter('walker_nav_menu_start_el', function ($item_output, $item, $depth, $args) {
    if (!empty($item->description)) {
        $item_output = str_replace($args->link_after . '</a>', '<span class="menu-item-description">' . $item->description . '</span>' . $args->link_after . '</a>', $item_output);
    }
    return $item_output;
}, 10, 4);`
It is not an Elementor filter, but a WordPress one. 
After enabling, customize with CSS.

**Reply by Caleforty** [reply]
Just stumbling on this while trying to get descriptions on mine. This code is getting it to display, but it displays inline with the nav title and same styling. Can't figure out how to crack overriding the Elementor styling with !important or function.php changes. Any suggestions on how to approach this?

**Reply by nmrelations** [reply]
This should work. :) Adjust the styling as before. 

`add_filter('walker_nav_menu_start_el', function ($item_output, $item, $depth, $args) {
    if (!empty($item->description)) {
        // Replaced inline with a separate line
        $item_output = str_replace($args->link_after . '</a>', '</a><span class="menu-item-description">' . $item->description . '</span>', $item_output);
    }
    return $item_output;
}, 10, 4);`
